sources/packages/backend/apps/api/src/route-controllers/education-program-offering/_tests_/e2e/education-program-offering.institutions.controller.updateProgramOffering.e2e-spec.ts

```diff
@@ -155,6 +160,179 @@ describe("EducationProgramOfferingInstitutionsController(e2e)-updateProgramOffer
+        showYearOfStudy: true,
+        offeringIntensity: OfferingIntensity.fullTime,
+        offeringDelivered: OfferingDeliveryOptions.Onsite,
+        hasOfferingWILComponent: "no",
+        studyStartDate: "2023-09-01",
+        studyEndDate: "2024-06-30",
+        lacksStudyBreaks: false,
+        studyBreaks: [
+          {
+            breakStartDate: studyBreak.breakStartDate,
+            breakEndDate: studyBreak.breakEndDate,
+          },
+        ],
+        offeringType: OfferingTypes.Public,
+        offeringDeclaration: true,
+        actualTuitionCosts: 1234,
+        programRelatedCosts: 3211,
+        mandatoryFees: 100001,
+        exceptionalExpenses: 555,
+      };
+
+      // Act/Assert
+      await request(app.getHttpServer())
+        .patch(endpoint)
+        .send(payload)
+        .auth(institutionUserToken, BEARER_AUTH_TYPE)
+        .expect(HttpStatus.OK)
+        .expect({});
+      const updatedEducationProgramOffering =
+        await db.educationProgramOffering.findOne({
+          where: { id: savedEducationProgramOffering.id },
+        });
+      expect(updatedEducationProgramOffering).toEqual(
+        expect.objectContaining({
+          name: payload.offeringName,
+          studyStartDate: payload.studyStartDate,
+          studyEndDate: payload.studyEndDate,
+          actualTuitionCosts: payload.actualTuitionCosts,
+          programRelatedCosts: payload.programRelatedCosts,
+          mandatoryFees: payload.mandatoryFees,
+          exceptionalExpenses: payload.exceptionalExpenses,
+          offeringDelivered: payload.offeringDelivered,
+          lacksStudyBreaks: payload.lacksStudyBreaks,
+          offeringType: payload.offeringType,
+          offeringIntensity: payload.offeringIntensity,
+          yearOfStudy: payload.yearOfStudy,
+          showYearOfStudy: payload.showYearOfStudy,
+          hasOfferingWILComponent: payload.hasOfferingWILComponent,
+          offeringWILType: null,
+          studyBreaks: {
+            totalDays: studyPeriodBreakdown.totalDays,
+            studyBreaks: [
+              {
+                breakStartDate: studyBreak.breakStartDate,
+                breakEndDate: studyBreak.breakEndDate,
+                breakDays: 32,
+                eligibleBreakDays: 21,
+                ineligibleBreakDays: 11,
+              },
+            ],
+            totalFundedWeeks: studyPeriodBreakdown.totalFundedWeeks,
+            fundedStudyPeriodDays: studyPeriodBreakdown.fundedStudyPeriodDays,
+            unfundedStudyPeriodDays:
+              studyPeriodBreakdown.unfundedStudyPeriodDays,
+          },
+          offeringDeclaration: payload.offeringDeclaration,
+          assessedDate: null,
+          offeringStatus: OfferingStatus.CreationPending,
+        }),
+      );
+    },
+  );
+
+  it(`Should return error when trying to update a new offering when data with exceptionalExpenses greater than ${MONEY_VALUE_FOR_UNKNOWN_MAX_VALUE} is passed.`, async () => {
```

Comment:
"Should return error when trying to update a new offering data with..." (two times "when")

Reply:
is that an issue? I have rephrased

Reply:
Not an issue. I just thought it would better for reading it.